tsdb/store.go

```diff
@@ -680,9 +730,13 @@ func (s *Store) deleteSeries(database string, seriesKeys []string, min, max int6
 		return influxql.ErrDatabaseNotFound(database)
 	}
 
-	for _, sh := range s.shards {
+	shards := s.filterShards(func(sh *Shard) bool {
+		return sh.database == database
+	})
+
+	return s.walkShards(shards, func(sh *Shard) error {
 		if sh.database != database {
-			continue
+			return nil
```

Comment:
If this ever executes, it either means that `filterShards` doesn't work, or `sh.database` changed since `filterShards` ran, making this check a race...